app/Models/Room.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Model;
+
+class Room extends Model
+{
+    protected $primaryKey = 'name';
+
+    public $incrementing = false;
+
+    protected $keyType = 'string';
+
+    public $timestamps = false;
+
+    protected $fillable = ['name', 'capacity'];
+
+    public function users()
+    {
+        return $this->belongsTo(User::class, 'name', 'room');
+    }
+
+    public function residentNumber()
+    {
+        return $this->users()->count();
+    }
+
+    public function color()
+    {
+        $color="#ffffff";
+        switch ($this->capacity-$this->residentNumber()) {
```

Comment:
??

Reply:
It assigns a color by how much capacity is left (maxCapacity-residentsAlreadyThere)

Reply:
Oh, then add spaces please :D

Reply:
And a comment